bg_mon.c

```diff
@@ -175,7 +217,7 @@ static void prepare_statistics_output(struct evbuffer *evb)
 			char *type = s.is_backend ? "\"backend\"" : ps.cmdline;
 			if (is_first) is_first = false;
 			else evbuffer_add_printf(evb, ", ");
-			evbuffer_add_printf(evb, "{\"pid\": %d, \"type\": %s, \"state\": \"%c\", ", s.pid, type, ps.state);
+			evbuffer_add_printf(evb, "{\"pid\": %d, \"type\": %s, \"state\": \"%c\", ", s.pid, type, ps.state ? ps.state : 'S');
```

Comment:
Just to clarify - S here stands for sleeping or something else?

Reply:
Yes, sleeping.

Reply:
What's the logic behind it? I mean, why if there's no status it's supposed to be sleeping?

Reply:
Information from procfs is collected independently from pg_stat_activity and there is a code which joins two lists of processes: https://github.com/CyberDem0n/bg_mon/blob/master/postgres_stats.c#L437-L456

It might happen that there was nothing in the procfs, but there was a row in pg_stat_activity.
We still want to display such a process, but some of the statistics for it would be empty.

In the case of `state`, we don't want to output '\0', but 'S', otherwise it breaks json